app/src/main/java/com/lucas/medicaltools/MainPresenter.kt

```diff
@@ -0,0 +1,23 @@
+package com.lucas.medicaltools
+
+import com.hannesdorfmann.mosby3.mvi.MviBasePresenter
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.service.GetUserUseCase
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+class MainPresenter: MviBasePresenter<MainView, MainViewState>() {
+    override fun bindIntents() {
+
+        /*val medicalToolsState: Observable<MainViewState> = intent(MainView::getMedicalToolsIntent)
```

Comment:
Ensure to map all your intent from the view, switchmap to use the interactor, ensure that each switch map from an intent produces a ViewState.